remotes/docker/pusher.go

```diff
@@ -156,13 +162,18 @@ func (p dockerPusher) push(ctx context.Context, desc ocispec.Descriptor, ref str
 		resp.Body.Close()
 	}
 
+	var pusher func() (*pushWriter, error)
+
 	if isManifest {
 		putPath := getManifestPath(p.object, desc.Digest)
 		req = p.request(host, http.MethodPut, putPath...)
 		req.header.Add("Content-Type", desc.MediaType)
 	} else {
 		// Start upload request
 		req = p.request(host, http.MethodPost, "blobs", "uploads/")
+		if host.ChunkSize > 0 {
+			req.header.Add("Content-Length", "0")
```

Comment:
Could you add a comment to explain this?

Reply:
Thanks for the reply, it seems the distribution spec (https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pushing-a-blob-in-chunks) requires the `Content-Length: 0` request header when starting a chunked upload, but in fact it still works without it, tested with `distribution/distribution:2.8.3`, let's remove it first.